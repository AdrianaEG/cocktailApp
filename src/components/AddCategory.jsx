import React from 'react'
import { useState } from 'react'

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');
    
    const onSubmit = (e)=>{
        e.preventDefault();
        if (inputValue.trim().length<=1) return;
        setInputValue('');
        onNewCategory(inputValue.trim());
    }
    const onInputChange = (e)=>{
        setInputValue(e.target.value)
    }
    
  return (
    <form onSubmit={onSubmit} aria-label="form">
        <input 
            type="text" 
            placeholder='Search by ingredient'
            value={inputValue}
            onChange={onInputChange}
        />
    </form>
  )
}
