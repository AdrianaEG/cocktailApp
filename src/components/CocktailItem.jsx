import React from 'react'

export const CocktailItem = ({title, url, id, setOpenPopUp, setCocktailSelected}) => {

  const showPopUp = (value)=>{
    setOpenPopUp(value);
    setCocktailSelected(id);
  }

  return (
    <>
      <div className='card' onClick={()=>{showPopUp(true)}}>
          <p>{title}</p>
          <img src={url} alt={title} />
      </div>
    </>
  )
}
