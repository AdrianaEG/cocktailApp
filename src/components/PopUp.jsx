import React from 'react'
import { useEffect, useState } from 'react'
import { getCocktail } from '../helpers/getCocktail'

export const PopUp = ({id, setOpenPopUp, cocktailSelected}) => {

    const [cocktail, setCocktail] = useState([]);
    const [isLoading, setIsLoading] = useState(true);

    const getDrink = async()=>{
      const newCocktail = await getCocktail(cocktailSelected)
      setCocktail(newCocktail);
      setIsLoading(false);
    }
    useEffect(()=>{
        getDrink(cocktailSelected);
    }, [])
    

  return (
    <div className='popup'>
        {
            isLoading && (
                <img src="https://i0.wp.com/css-tricks.com/wp-content/uploads/2021/08/s_2A9C470D38F43091CCD122E63014ED4503CAA7508FAF0C6806AE473C2B94B83E_1627522653545_loadinfo.gif?resize=200%2C200&ssl=1" className="loading"/>
            )
        }
        <button onClick={()=>{setOpenPopUp(false)}}>BACK</button>
        <h1>{cocktail.title}</h1>
        <p className='m-0 w-bold'>Glass: <span>{cocktail.glass}</span></p>
        <p className='m-0 w-bold'> Category: <span>{cocktail.category}</span></p>
        <img src={cocktail.image} alt={cocktail.title}/>
        <p className='mt-0 w-bold'>Ingredients:
        {
            cocktail.ingredients && (
                cocktail.ingredients.map((ingredient)=>(
                    ingredient&&<span> {ingredient}. </span>
                ))
            )
        }
        </p>
        <p className='mt-0 w-bold' >Instructions: <span>{cocktail.instructions}</span></p>
        {cocktail.video && <link rel="stylesheet" href={cocktail.video} />}
    </div>
    
  )
}
