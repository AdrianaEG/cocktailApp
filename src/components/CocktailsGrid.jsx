import React from 'react';
import { CocktailItem } from './CocktailItem';
import { useFetchCocktails } from '../hooks/useFetchCocktails';


export const CocktailsGrid = ({category, setOpenPopUp, setCocktailSelected}) => {

  const {cocktails, isLoading, error } = useFetchCocktails(category)

  return (
    <>
    {
      error ?
      <>
        <p className='warning'>No se encontraron cocteles que contengan {category}</p>
        
      </>
      :
      <>
        <h3>{category}</h3>
        {
          isLoading &&
          <img src="https://i0.wp.com/css-tricks.com/wp-content/uploads/2021/08/s_2A9C470D38F43091CCD122E63014ED4503CAA7508FAF0C6806AE473C2B94B83E_1627522653545_loadinfo.gif?resize=200%2C200&ssl=1" alt="cargando..." className='loading' />
        }
                <div className='card-grid'>
          { cocktails && cocktails.map((cocktail)=>(
            <CocktailItem 
              key={cocktail.id} 
              title={cocktail.title} 
              url={cocktail.url} 
              id={cocktail.id}
              setOpenPopUp={setOpenPopUp}
              setCocktailSelected={setCocktailSelected}
              />
          ))}
         
        </div>
      </> 
    }
    </>
  )
}
