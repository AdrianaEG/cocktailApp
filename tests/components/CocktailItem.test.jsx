import { CocktailItem } from "../../src/components/CocktailItem"
import {render, screen } from '@testing-library/react';

describe('Pruebas en el componente CocktailItem', ()=>{
    const title="party";
    const url="https://party.com/"
    test('Debe hacer match con el snapshot', ()=>{
        const {container} = render(<CocktailItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })
    test('El componente CocktailItem debe si o si tener una url y un título', ()=>{
        render(<CocktailItem title={title} url={url}/>)
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })
    test('Debe mostrar el título en el componente', ()=>{
        render(<CocktailItem title={title} url={url}/>)
       expect(screen.getByText(title)).toBeTruthy();
    })
})