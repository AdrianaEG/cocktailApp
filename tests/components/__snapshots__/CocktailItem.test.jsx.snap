// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente CocktailItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <p>
      party
    </p>
    <img
      alt="party"
      src="https://party.com/"
    />
  </div>
</div>
`;
